SimPEG/electromagnetics/natural_source/receivers.py

```diff
@@ -29,318 +31,77 @@ class BaseRxNSEM_Point(BaseRx):
         {
             "real": ["re", "in-phase", "in phase"],
             "imag": ["imaginary", "im", "out-of-phase", "out of phase"],
+            "apparent resistivity": [
+                "apparent_resistivity",
+                "apparent-resistivity",
+                "app_rho",
+            ],
+            "phase": ["phi"],
         },
     )
 
-    def __init__(self, locs, orientation=None, component=None):
+    def __init__(
+        self,
+        locations=None,
+        orientation=None,
+        component=None,
+        locations_e=None,
+        locations_b=None,
+    ):
         self.orientation = orientation
         self.component = component
+        self.locations_e = locations_e
+        self.locations_b = locations_b
 
-        BaseRx.__init__(self, locs)
-
-    # Set a mesh property - TODO: remove the following properties
-    @property
-    def mesh(self):
-        return self._mesh
-
-    @mesh.setter
-    def mesh(self, value):
-        if value is getattr(self, "_mesh", None):
-            pass
-        else:
-            self._mesh = value
-
-    @property
-    def src(self):
-        return self._src
+        if locations is None:
+            locations = np.hstack([locations_e, locations_b])
 
-    @src.setter
-    def src(self, value):
-        self._src = value
-
-    @property
-    def f(self):
-        return self._f
-
-    @f.setter
-    def f(self, value):
-        self._f = value
+        BaseRx.__init__(self, locations)
 
     def _locs_e(self):
-        if self.locations.ndim == 3:
-            loc = self.locations[:, :, 0]
+        if self.locations_e is None:
```

Comment:
do we still need `_locs_e` or should we just put this logic in the `__init__` and work only with `locations`, `locations_e`, `locations_h` (or `b` depending on what we decide to go with)

Reply:
Just to make sure I follow; we strictly use locations_e & locations_h but if initiated with locations, the initialization will parse it into locations_e and locations_b, correct?

Reply:
currently, we have 
```
def __init__(
        self,
        locations=None,
        orientation=None,
        component=None,
        locations_e=None,
        locations_b=None,
    ):
        self.orientation = orientation
        self.component = component
        self.locations_e = locations_e
        self.locations_b = locations_b

        if locations is None:
            locations = np.hstack([locations_e, locations_b])

        BaseRx.__init__(self, locations)

    def _locs_e(self):
        if self.locations_e is None:
            if self.locations.ndim == 3:
                loc = self.locations[:, :, 0]
            else:
                loc = self.locations
            return loc
        else:
            return self.locations_e

    def _locs_h(self):
        if self.locations_b is None:
            if self.locations.ndim == 3:
                loc = self.locations[:, :, 1]
            else:
                loc = self.locations
            return loc
        else:
            return self.locations_b
```

What I think we should do is something like 
```
def __init__(
        self,
        locations=None,
        orientation=None,
        component=None,
        locations_e=None,
        locations_b=None,
    ):
        self.orientation = orientation
        self.component = component

        # check that either locations_e and locations_b provided or locations provided and error if not
        
       if locations_e, locations_b provided: 
            self._locations_e = locations_e
            self._locations_b = locations_b
        
        elif locations is not None: 
            # check shape of locations, 
            if width 2 or a list of length 2: 
                  self._locations_e = locations[:, 0]
                  self._locations_b = locations[:, 1]
            if width 1:
                  self._locations_e = locations
                  self._locations_b = locations
            
        locations = np.hstack([self._locations_e, self._locations_b])

        BaseRx.__init__(self, locations)

    def locations_e(self):
           return self._locations_e

    def locations_b(self): # or h
            return self._locations_b
```

Reply:
There should also be a check to ensure they're the same length.